st2actions/st2actions/utils/param_utils.py

```diff
@@ -215,15 +215,25 @@ def _do_render_params(renderable_params, context):
     env = Environment(undefined=StrictUndefined)
     rendered_params = {}
     rendered_params.update(context)
+
+    keys_to_delete = []
     while len(renderable_params) != 0:
         for k, v in six.iteritems(renderable_params):
             template = env.from_string(v)
+
             try:
                 rendered = template.render(rendered_params)
                 rendered_params[k] = rendered
-            except:
+                keys_to_delete.append(k)
+            except Exception:
+                # TODO: Maybe we should throw here and abort action execution?
```

Comment:
I went ahead and made the throw change.

Previously, I just wanted to follow the existing flow and not change the logic, but the existing logic was broken anyway (we ended up in an infinite loop).

Now if rendering a param fail, we throw an exception which causes action to fail and the error is reported to the user.